### PR TITLE
Adjust the test suite for Windows

### LINKED ISSUE #39: Tests fail on Windows!
I hate windows!

When you run `stack test` on the fresh clone of commit c9e3aae8 in a windows 10 environment, you'll see (at compile time !!) the build will fail:

Not sure yet what's going on, but will address soon!

### PR DESCRIPTION
**Pull request by mrkkrp**
Close #39.

### CODE REVIEW

**Comment by mrkkrp** [comment]
Actually looks like we need two separate tests suites, one for Unix and the other one for Windows, because path format is quite different on these systems.

**Comment by mrkkrp** [comment]
Does Window allow to use `/` as path separator?

**Comment by sjakobi** [comment]
> Does Window allow to use / as path separator?

[Yes!](https://github.com/haskell/filepath/blob/141cddb606fd6f6a60f730fed3d57502b93c14ae/System/FilePath/Internal.hs#L157)

**Comment by mrkkrp** [comment]
Valitity test seems to fail:

https://travis-ci.org/commercialhaskell/path/jobs/208374928

**Comment by sjakobi** [comment]
> Valitity test seems to fail:

Please deactivate them for now. I'll look into this.

**Comment by sjakobi** [comment]
BTW I just noticed that you can import and use `System.FilePath.Windows` on Linux. Thought this might be handy for you.

**Comment by mrkkrp** [comment]
Yep, this will help me with trying it out. 👍 I plan to finish the PR this evening.

**Comment by mrkkrp** [comment]
OK, this should pass. I commented out about 10 cases which fail and I don't know if this should be considered bugs or something. I marked these with `FIXME`s.

File: stack.yaml [hunk 1]
```diff
@@ -1,6 +1,3 @@
-flags:
-  path:
-    validity: true
```
Comment (sjakobi) [inline stack.yaml]: Can you add back the flag and instead use the `:test` target in the appveyor.yml? This way validity-test should still run on Travis.

**Comment by sjakobi** [comment]
From a cursory look you've only used the basic `C:\\` drive syntax for absolute paths so far. Sadly there's a lot more [out there](http://hackage.haskell.org/package/filepath-1.4.1.2/docs/System-FilePath-Windows.html#v:splitDrive), but I don't see why we shouldn't start with this and add the rest later.

I haven't looked at the FIXMEs so far.

File: stack.yaml (hunk 1 above)
Comment (mrkkrp) [inline stack.yaml]: But it failed to Travis!

Comment (sjakobi) [inline stack.yaml]: Since https://github.com/commercialhaskell/path/pull/78 it shouldn't any more.

**Comment by mrkkrp** [comment]
@sjakobi How does it look now?

**Comment by sjakobi** [comment]
Looks like you should remove the  `--no-run-tests` flag on AppVeyor! ;)

**Comment by mrkkrp** [comment]
I thought it's only for validity stuff... We have it on master.

**Comment by mrkkrp** [comment]
Removed `--no-run-tests` and it fails now.

File: appveyor.yml [hunk 2]
```diff
@@ -18,4 +18,4 @@ test_script:
 - stack setup > nul
 # The ugly echo "" hack is to avoid complaints about 0 being an invalid file
 # descriptor
-- echo "" | stack --no-terminal build --haddock --no-haddock-deps --bench --test :validity-test --no-run-tests
+- echo "" | stack --no-terminal build --haddock --no-haddock-deps --bench --test :validity-test
```
Comment (sjakobi) [inline appveyor.yml]: > Removed --no-run-tests and it fails now.

`:validity-test` is still broken on Windows (#74). As you're working on `:test` here, you'll want to test `:test`. :)

Comment (mrkkrp) [inline appveyor.yml]: Thanks for clarification, my bad. Somehow I didn't think of `:validity-test` as argument of `--test`. I usually run `stack test` and it runs all the tests. Still too lazy to learn more!

**Comment by mrkkrp** [comment]
OK, this should be good to go, finally. Still we need to decide about the `FIXME`s.

**Pull request merged**